src/plugins/coreplugin/iversioncontrol.h

```diff
@@ -36,10 +36,15 @@
 #include <QObject>
 #include <QString>
 
+#include <tuple>
+
 QT_FORWARD_DECLARE_CLASS(QMenu);
 
 namespace Core {
 
+using ChangeSet = QSet<QString>;
+using ChangeSets = std::tuple<ChangeSet, ChangeSet>;
```

Comment:
Can we use a struct here, to make clear that we talk about tracked and untracked changes?
Or might it be better to have a hash from filename to an enum of change type?

Reply:
> Do you mean like this?
> 
> ```cpp
> enum class ChangeType {
> Tracked,
> Untracked
> };
> using Changes = QHash<QString, ChangeType>;
> ```
Yes, looks good. But I guess it depends if this simplifies other code or not.